src/main/java/com/pnu/dev/radioserviceapi/config/WebSecurityConfig.java

```diff
@@ -0,0 +1,53 @@
+package com.pnu.dev.radioserviceapi.config;
+
+import lombok.SneakyThrows;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+
+@Configuration
+@EnableWebSecurity
+public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+
+    @Value("${user.username}")
+    private String USERNAME;
+
+    @Value("${user.password}")
+    private String PASSWORD;
+
+    @Override
+    protected void configure(HttpSecurity httpSecurity) throws Exception {
+        httpSecurity
+                //Admin security
+                .authorizeRequests()
+                .antMatchers("/admin/**")
+                .authenticated()
+                .and()
+                .formLogin()
+                .loginPage("/login")
+                .defaultSuccessUrl("/admin")
+                .permitAll()
+                .and()
+                .logout()
+                .permitAll()
+                //API security
+                .and()
+                .authorizeRequests()
+                .antMatchers("/api/**")
+                .permitAll();
+    }
+
+    @Autowired
+    @SneakyThrows
+    public void configureGlobal(AuthenticationManagerBuilder auth) {
+        auth.inMemoryAuthentication()
+                .withUser(USERNAME)
+                .password(String.format("{noop}%s", PASSWORD))
```

Comment:
Just wonder, why do we need this `{noop}` at the beginning?

Reply:
To command `AuthenticationManagerBuilder ` not to encode password